functions.php

```diff
@@ -0,0 +1,15 @@
+<?php
+function generateCharacterHtml(array $allCharacters): string {
+    if(count($allCharacters) == count($allCharacters, COUNT_RECURSIVE)){
+        return "Incorrect input! Try again:-)";
+    }
+    $characterHtml = "";
+    foreach ($allCharacters as $character) {
+        $characterHtml .= '<div class="containerItem">' . '<p>' . 'Name: ' . $character['name'] . '</p>' .
+            '<p>' . 'Alignment: ' . $character['alignment'] . '</p>' .
```

Comment:
no need to concatenate `'<p>' . 'Alignment'` - just write `<p>Alignment` (and same in a lot of places)

Reply:
still some little concatenations left in the closing tags
```php
$characterHtml .= '<div class="individualCharacter">' . '<p>' . 'Name: ' . $character['name'] . '</p>' .
            '<p>Alignment: ' . $character['alignment'] . '</p>' .
            '<p>Weight: ' . $character['weight_kg'] . 'kg' . '</p>' . // here
            '<p>Height: ' . $character['height_cm'] . 'cm' . '</p>' . // here
            '<p>Place of Birth: ' . $character['place_of_birth'] . '</p>' . '</div>';
```